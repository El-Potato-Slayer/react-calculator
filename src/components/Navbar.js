import { Link } from 'react-router-dom';

function Navbar() {
  return (
    <nav data-testid="navbar" className="navbar">
      <p>Mathematics Meow Meow</p>
      <div>
        <Link to="/">Home</Link>
        <Link to="/calculator">Calculator</Link>
        <Link to="/quote">Quote Of The Day</Link>
      </div>
    </nav>
  );
}

export default Navbar;
