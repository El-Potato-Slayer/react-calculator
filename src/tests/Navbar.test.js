import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

afterEach(cleanup);

it('Renders Navbar component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
    div,
  );
});

it("ensures 'Mathematics Meow Meow' text is rendered", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(getByTestId('navbar')).toHaveTextContent('Mathematics Meow Meow');
});

it("ensures 'Home' link is rendered", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(getByTestId('navbar')).toHaveTextContent('Home');
});

it("ensures 'Calculator' link is rendered", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(getByTestId('navbar')).toHaveTextContent('Calculator');
});

it("ensures 'Quote Of The Day' link is rendered", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(getByTestId('navbar')).toHaveTextContent('Quote Of The Day');
});

it("ensures 'Swag' text is not rendered", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(getByTestId('navbar')).not.toHaveTextContent('Swag');
});

it('matches snapshot', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
