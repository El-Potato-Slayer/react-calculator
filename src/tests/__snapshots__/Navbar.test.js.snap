// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar"
  data-testid="navbar"
>
  <p>
    Mathematics Meow Meow
  </p>
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      href="/quote"
      onClick={[Function]}
    >
      Quote Of The Day
    </a>
  </div>
</nav>
`;
